src/test/java/de/gwdg/metadataqa/marc/utils/marcspec/legacy/MarcSpecTest.java

```diff
@@ -146,16 +147,16 @@ public void testEncode() {
   @Test
   public void testValidity() {
     MarcSpec marcSpec = new MarcSpec();
-    assertTrue(marcSpec.validate("245"));
-    assertTrue(marcSpec.validate("245ab"));
-    assertTrue(marcSpec.validate("XXXab"));
-    assertTrue(marcSpec.validate("245ab_1"));
-    assertTrue(marcSpec.validate("245ab__0"));
-    assertTrue(marcSpec.validate("245ab_10"));
-    assertTrue(marcSpec.validate("245ab_1_"));
-    assertTrue(marcSpec.validate("245$a"));
-    assertTrue(marcSpec.validate("LDR/0-4"));
-    assertTrue(marcSpec.validate("880_1"));
+    marcSpec.validate("245");
```

Comment:
Why the assertion is removed from here?

Reply:
The method used to return `true` all the time. The only way the validation would fail wasn't to return `false`, but to throw an exception. Considering that it's validation that's the main focus here, I was thinking the results should be either `true`-`false` or *exception*-*no exception* and I opted in for the second one given that the cases where the validation fails are more important and implemented that way already, so I removed the unnecessary `return true` statement.

The test fails in case the failing exception gets thrown, so only the syntactical approach has changed because `assertTrue` method would never get a `false` value.

Reply:
@gegic Thanks for the explanation! I'd better to keep the assertion even if false is never returned. BTW: this part of the tool is a Java implementation of a PHP code, I plan to revise it in the future.